.github/workflows/reusable-ubsan.yml

```
@@ -0,0 +1,74 @@
+name: Reusable Undefined Behavior Sanitizer
+
+on:
+  workflow_call:
+    inputs:
+      config_hash:
+        required: true
+        type: string
+
+env:
+  FORCE_COLOR: 1
+
+jobs:
+  build-ubsan-reusable:
+    name: 'Undefined behavior sanitizer'
+    runs-on: ubuntu-24.04
+    timeout-minutes: 60
+    steps:
+    - uses: actions/checkout@v4
+      with:
+        persist-credentials: false
+    - name: Runner image version
+      run: echo "IMAGE_OS_VERSION=${ImageOS}-${ImageVersion}" >> "$GITHUB_ENV"
+    - name: Restore config.cache
+      uses: actions/cache@v4
+      with:
+        path: config.cache
+        key: ${{ github.job }}-${{ env.IMAGE_OS_VERSION }}-${{ inputs.config_hash }}
```

I think the configure failure:

https://github.com/python/cpython/actions/runs/15706865183/job/44254688889?pr=135578

is because this key is identical to the one in `reusable-tsan.yml`, so let's add `ubsan` or something in here:

```suggestion
        key: ${{ github.job }}-ubsan-${{ env.IMAGE_OS_VERSION }}-${{ inputs.config_hash }}
```

---

That's part of `github.job`. It's using a cache from before I added the latest commits.
I removed that cache and will restart the job. (GitHub will allow that after all the other ones finish).